### PR TITLE
Stitch AI - Initial POC

### PR DESCRIPTION
**Pull request by narner**
Entering OpenAI API Key:

CleanShot 2024-10-10 at 17 13 04@2x

Entering a user prompt:

CleanShot 2024-10-10 at 17 13 21@2x

Displaying alerts with full context

CleanShot 2024-10-14 at 09 28 24@2x

### CODE REVIEW

**Review by pianostringquartet — APPROVED** [review]
great stuff; the error messages for API key was helpful as well, I had messed up my key initially.

I noticed that it took chat GPT a while to generate this graph from the "display 4 triangles" prompt I gave it; initially I thought some silent error had happened. 

Should we show a "Generating Graph" message while we're waiting for OpenAI's response ?
Or maybe e.g. don't close the modal until we get back a response? 
Maybe that's not good for timeouts ?

Screenshot 2024-10-14 at 8 36 48 AM

**Comment by pianostringquartet** [comment]
Also, all the nodes started out right on top of each other; might be good to at least stagger them?

File: Stitch/Graph/StitchAI/StitchAIDataModel.swift [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    case text = "Text"
+    case oval = "Oval"
+    case rectangle = "Rectangle"
+    case shape = "Shape"
+    case colorFill = "ColorFill"
+    case image = "Image"
+    case video = "Video"
+    case videoStreaming = "VideoStreaming"
+    case realityView = "RealityView"
+    case canvasSketch = "CanvasSketch"
```
[the remark this answers is not in the document]
Comment (narner) [reply Stitch/Graph/StitchAI/StitchAIDataModel.swift:248]: Yeah; doesn't really matter
We'll clean up this logic to just use what we have in the app later

**Comment by narner** [comment]
Updated with status text: 

https://github.com/user-attachments/assets/48e491fd-1fd7-4941-8135-8c5b903023c8

**Comment by narner** [comment]
> Also, all the nodes started out right on top of each other; might be good to at least stagger them? e.g. as we're processing the json / array of LLM actions, keep a running tab of how many nodes have been added and increment it and place nodes at `position(x: nodesAddedSoFar * 300, y: nodesAddedSoFar * 300)`

Good idea; but will handle this after we've gotten further along

**Pull request merged**